Dask distributed integration tests are failing on master

For example:
```
_______________ test_dask_distributed_integration_test[netcdf4] ________________
loop = <tornado.platform.epoll.EPollIOLoop object at 0x7f4c45739710>
engine = 'netcdf4'
    @pytest.mark.parametrize('engine', ENGINES)
    def test_dask_distributed_integration_test(loop, engine):
        with cluster() as (s, _):
>           with distributed.Client(('127.0.0.1', s['port']), loop=loop):
E           KeyError: 'port'
```

@mrocklin did you change the API for this?

--- a/xarray/tests/test_distributed.py
+++ b/xarray/tests/test_distributed.py
@@ -28,7 +28,7 @@ def test_dask_distributed_integration_test(loop, engine):
             original = create_test_data()
             with create_tmp_file() as filename:
                 original.to_netcdf(filename, engine=engine)
-                restored = xr.open_dataset(filename, chunks=3, engine=engine)
-                assert isinstance(restored.var1.data, da.Array)
-                computed = restored.compute()
-                assert_allclose(original, computed)
+                with xr.open_dataset(filename, chunks=3, engine=engine) as restored:
+                    assert isinstance(restored.var1.data, da.Array)
+                    computed = restored.compute()
+                    assert_allclose(original, computed)

--- a/xarray/tests/test_distributed.py
+++ b/xarray/tests/test_distributed.py
@@ -6,7 +6,7 @@ distributed = pytest.importorskip('distributed')
 da = pytest.importorskip('dask.array')
 from distributed.utils_test import cluster, loop
 
-from xarray.tests.test_backends import create_tmp_file
+from xarray.tests.test_backends import create_tmp_file, ON_WINDOWS
 from xarray.tests.test_dataset import create_test_data
 
 from . import assert_allclose, has_scipy, has_netCDF4, has_h5netcdf
@@ -26,7 +26,7 @@ def test_dask_distributed_integration_test(loop, engine):
     with cluster() as (s, _):
         with distributed.Client(s['address'], loop=loop):
             original = create_test_data()
-            with create_tmp_file() as filename:
+            with create_tmp_file(allow_cleanup_failure=ON_WINDOWS) as filename:
                 original.to_netcdf(filename, engine=engine)
                 with xr.open_dataset(filename, chunks=3, engine=engine) as restored:
                     assert isinstance(restored.var1.data, da.Array)

--- a/xarray/tests/test_distributed.py
+++ b/xarray/tests/test_distributed.py
@@ -24,7 +24,7 @@ if has_h5netcdf:
 @pytest.mark.parametrize('engine', ENGINES)
 def test_dask_distributed_integration_test(loop, engine):
     with cluster() as (s, _):
-        with distributed.Client(('127.0.0.1', s['port']), loop=loop):
+        with distributed.Client(s['address'], loop=loop):
             original = create_test_data()
             with create_tmp_file() as filename:
                 original.to_netcdf(filename, engine=engine)

